plugin/path_role.go

```diff
@@ -288,8 +288,8 @@ func (b *GcpAuthBackend) pathRoleRead(ctx context.Context, req *logical.Request,
 
 	resp := make(map[string]interface{})
 
-	if role.RoleType != "" {
-		resp["role_type"] = role.RoleType
+	if role.Type != "" {
+		resp["type"] = role.Type
```

Comment:
A quicker fix is to undo all and just change this string value "role_type" to "type". I'd prefer this, just because there are fewer places where you could possibly miss a reference to role_type/RoleType. If you can also change https://github.com/hashicorp/vault-plugin-auth-gcp/blob/master/plugin/path_role_test.go and run the test (`make test TEST=./plugin`) that would be helpful! Thank you.

Reply:
Also, possibly more importantly, this avoids errors when reading role structs/JSON stored in Vault storage from previous versions of the plugin.

Reply:
Ah, good call. I originally did this but talked myself out of it. I'll update the PR!